rbac/management/role/view.py

```diff
@@ -349,8 +343,27 @@ def list(self, request, *args, **kwargs):
             "username": request.query_params.get("username", None),
             "limit": request.query_params.get("limit", 10),
             "offset": request.query_params.get("offset", 0),
-            "add_fields": request.query_params.get("add_fields", None),
+            "add_fields": request.query_params.get("add_fields", ""),
         }
+        add_fields = query_params["add_fields"]
+        add_fields_split = add_fields.split(",")
+
+        base_queryset = (
+            Role.objects.prefetch_related("access", "ext_relation")
+            .filter(tenant__in=[request.tenant, public_tenant])
+            .annotate(
+                policyCount=Count("policies", distinct=True),
+                accessCount=Count("access", distinct=True),
+            )
+        )
+
+        # Dynamic annotations
+        if "groups_in_count" in add_fields_split:
```

Comment:
I think the groups_in_count, groups_in, or without are not mutual exclusive, so no if else needed here.

And obtain_groups_in might still needed here? There are some logic about scope and admin has to be included there